src/Microsoft.AspNetCore.Dispatcher/Patterns/RoutePatternBuilder.cs

```diff
@@ -71,6 +71,11 @@ public RoutePattern Build()
 
         public static RoutePatternBuilder Create(string text)
         {
+            if (text.StartsWith("~/"))
+            {
+                return new RoutePatternBuilder() { RawText = text.Substring(1), };
+            }
```

Comment:
The `RawText` should still reflect what the user typed in. I don't think we should have this special case.